server/routes/pubs.js

```diff
@@ -0,0 +1,53 @@
+const express = require('express');
+const router = express.Router();
+// const { ApiService } = require('../services');
+const Pub = require('../database/models/pub');
+
+
+router.get('/data', (req, res, next) => {
+    Pub.find().sort({ 'name': 1 })
+        .then(function (dbPubs) {
+            res.json(dbPubs);
+        })
+    // (error) => res.sendStatus(404)
+    // );
+});
+
+router.post('/', (req, res, next) => {
+    Pub.findOne({ twitterHandle: req.body.twitterHandle }, (err, pub) => {
+        if (err) {
+            console.log("here is the error: ", err)
+        } else if (pub) {
+            alert('That publication is already in the database')
+            window.location.replace('/pub-display');
+        } else {
+            Pub.create({
+                name: req.body.name,
+                twitterHandle: req.body.twitterHandle
+            })
+                .then(function (dbPub) {
+                    res.json(dbPub);
+                });
+        }
+    })
+});
+
+router.get('/searchPubTitle/:title', (params, res, next) => {
+    var regex = new RegExp(params.params.title, "i");
+    Pub.find({ name: regex })
+        .then(function (dbTitle) {
+            res.json(dbTitle);
+        });
+})
+
+router.get('/searchPubTwitter/:twitter', (params, res, next) => {
+    var regex = new RegExp(["^", params.params.title, "$"].join(""), "i");
```

Comment:
This search does not appear to work for partial matches. I think it's because start/end of string is specified with `^` and `$`. It think we also mean `params.params.twitter` here.

Reply:
Fixed both issues, and search works correctly now.